leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -312,6 +312,37 @@ public boolean stepDown() {
         return false;
     }
 
+    @Override
+    public boolean hostileTakeover() {
+        LeadershipState leadershipState = determineLeadershipState();
+        StillLeadingStatus status = leadershipState.status();
+        switch (status) {
+            case LEADING:
+                return true;
+            case NOT_LEADING:
+                try {
+                    proposer.propose(getNextSequenceNumber(leadershipState.greatestLearnedValue()), null);
```

Comment:
nit: the idea that we propose `null` is now baked in to three places in the code (propose, stepdown and here) - maybe we should refactor this slightly?

Reply:
can do